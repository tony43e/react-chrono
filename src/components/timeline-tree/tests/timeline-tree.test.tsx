import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import {
  TimelineItemModel,
  TimelineItemViewModel,
} from "../../models/TimelineItemModel";
import Tree from "../index";

const onClick = jest.fn();
const autoScroll = jest.fn();

const items: TimelineItemModel[] = [
  {
    id: "1",
    title: "June 28, 1914 June 28, 1914 June 28, 1914June 28, 1914",
    contentText:
      "Archduke Franz Ferdinand of Austria and his wife, Sophie, are assassinated by a Bosnian Serb nationalist in Sarajevo.",
  },
  {
    id: "2",
    title: "July 28, 1914",
    contentText:
      "World War I begins when Austria-Hungary declares war on Serbia.",
  },
  {
    id: "3",
    title: "August 1–28, 1914",
    contentText: `Germany declares war on Russia, France, and Belgium. Britain declares war on Germany.
    Austria declares war on Russia. Montenegro declares war on Austria. France declares war on Austria.
    Britain declares war on Austria. Montenegro declares war on Germany. Japan declares war on Germany. Austria declares war on Belgium.`,
  },
  {
    id: "4",
    title: "September 6, 1914",
    contentText: `First Battle of the Marne begins. The Germans had advanced to within 30 miles of Paris, but over the next two days, the French are reinforced by 6,000 infantrymen who are transported to the front by hundreds of taxis.
      The Germans dig in north of the Aisne River, and the trench warfare that is to typify the Western Front for the next four years begins.`,
  },
];

const View = (
  <Tree
    onClick={onClick}
    autoScroll={autoScroll}
    items={items as TimelineItemViewModel[]}
    activeTimelineItem={2}
  />
);

test("Test Tree render", () => {
  render(View);
  const element = screen.getByTestId("tree-main");

  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

test("Test Tree Items", () => {
  render(View);
  const element = screen.getByTestId("tree-main");

  expect(element.children.length).toEqual(4);
});

test("Test Tree Branches", () => {
  render(View);
  const element = screen.getByTestId("tree-main");

  Array.from(element.children).forEach((child, index) => {
    if (index % 2 === 0) {
      expect(child).toHaveClass("left");
    } else {
      expect(child).toHaveClass("right");
    }
  });
});

// test("Test Tree Branch active", () => {
//   render(View);
//   const element = screen.getByTitle(/August 1–28, 1914/i);
//   expect(element).toHaveClass("active");
// });

test("Test Tree Branch onClick", () => {
  render(View);
  const element = screen.getAllByTestId("tree-leaf-click")[0];
  fireEvent.click(element);
  expect(onClick).toBeCalled();
  expect(onClick).toBeCalledWith("1");
});
