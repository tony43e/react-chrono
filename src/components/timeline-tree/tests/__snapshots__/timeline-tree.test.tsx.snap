// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tree render 1`] = `
<div
  class="sc-fzozJi iNxIKn"
  data-testid="tree-main"
>
  <div
    class="sc-fzoLsD dxSmZJ left"
    data-testid="branch-main"
  >
    <div
      class="sc-fznZeY jzOIvf left"
    >
      <div
        class="sc-AxiKw jUuDCK"
        title="June 28, 1914 June 28, 1914 June 28, 1914June 28, 1914"
      >
        June 28, 1914 June 28, 1914 June 28, 1914June 28, 1914
      </div>
    </div>
    <div
      class="sc-fznyAO hhDdtW left"
    >
      <div
        class="sc-AxjAm eEVtq"
      >
        <span
          class="sc-AxirZ izgUJV"
        >
          Archduke Franz Ferdinand of Austria and his wife, Sophie, are assassinated by a Bosnian Serb nationalist in Sarajevo.
        </span>
      </div>
    </div>
    <div
      class="sc-fzplWN hftWTh left"
      data-testid="tree-leaf"
    >
      <div
        class="sc-fznKkj dpatIe left"
        data-testid="tree-leaf-click"
      >
        <div
          class="sc-AxheI dRuAGA in-active"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-fzoLsD dxSmZJ right"
    data-testid="branch-main"
  >
    <div
      class="sc-fznZeY jzOIvf right"
    >
      <div
        class="sc-AxiKw jUuDCK"
        title="July 28, 1914"
      >
        July 28, 1914
      </div>
    </div>
    <div
      class="sc-fznyAO hhDdtW right"
    >
      <div
        class="sc-AxjAm eEVtq"
      >
        <span
          class="sc-AxirZ izgUJV"
        >
          World War I begins when Austria-Hungary declares war on Serbia.
        </span>
      </div>
    </div>
    <div
      class="sc-fzplWN hftWTh right"
      data-testid="tree-leaf"
    >
      <div
        class="sc-fznKkj dpatIe right"
        data-testid="tree-leaf-click"
      >
        <div
          class="sc-AxheI dRuAGA in-active"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-fzoLsD dxSmZJ left"
    data-testid="branch-main"
  >
    <div
      class="sc-fznZeY jzOIvf left"
    >
      <div
        class="sc-AxiKw jUuDCK active"
        title="August 1–28, 1914"
      >
        August 1–28, 1914
      </div>
    </div>
    <div
      class="sc-fznyAO hhDdtW left"
    >
      <div
        class="sc-AxjAm eEVtq active"
      >
        <span
          class="sc-AxirZ izgUJV"
        >
          Germany declares war on Russia, France, and Belgium. Britain declares war on Germany.
    Austria declares war on Russia. Montenegro declares war on Austria. France declares war on Austria.
    Britain declares war on Austria. Montenegro declares war on Germany. Japan declares war on Germany. Austria declares war on Belgium.
        </span>
      </div>
    </div>
    <div
      class="sc-fzplWN hftWTh left"
      data-testid="tree-leaf"
    >
      <div
        class="sc-fznKkj dpatIe left"
        data-testid="tree-leaf-click"
      >
        <div
          class="sc-AxheI dRuAGA active"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-fzoLsD dxSmZJ right"
    data-testid="branch-main"
  >
    <div
      class="sc-fznZeY jzOIvf right"
    >
      <div
        class="sc-AxiKw jUuDCK"
        title="September 6, 1914"
      >
        September 6, 1914
      </div>
    </div>
    <div
      class="sc-fznyAO hhDdtW right"
    >
      <div
        class="sc-AxjAm eEVtq"
      >
        <span
          class="sc-AxirZ izgUJV"
        >
          First Battle of the Marne begins. The Germans had advanced to within 30 miles of Paris, but over the next two days, the French are reinforced by 6,000 infantrymen who are transported to the front by hundreds of taxis.
      The Germans dig in north of the Aisne River, and the trench warfare that is to typify the Western Front for the next four years begins.
        </span>
      </div>
    </div>
    <div
      class="sc-fzplWN hftWTh right"
      data-testid="tree-leaf"
    >
      <div
        class="sc-fznKkj dpatIe right"
        data-testid="tree-leaf-click"
      >
        <div
          class="sc-AxheI dRuAGA in-active"
        />
      </div>
    </div>
  </div>
</div>
`;
